Development/nmos/configuration_api.cpp

```diff
@@ -634,6 +647,97 @@ namespace nmos
             });
         });
 
+        configuration_api.support(U("/rolePaths/") + nmos::patterns::rolePath.pattern + U("/bulkProperties/?"), methods::GET, [&model, get_properties_by_path_method_handler, &gate_](http_request req, http_response res, const string_t&, const route_parameters& parameters)
+        {
+            const auto role_path = parameters.at(nmos::patterns::rolePath.name);
+
+            auto lock = model.read_lock();
+            auto& resources = model.control_protocol_resources;
+
+            const auto& resource = details::find_resource(resources, role_path);
+            if (resources.end() != resource)
+            {
+                using web::json::value_of;
+
+                bool recurse = details::parse_recurse_query_parameter(req.request_uri().query());
+
+                web::json::value arguments = value_of({
+                        { nmos::fields::nc::recurse, recurse }
+                    });
+
+                auto result = get_properties_by_path(resources, *resource, arguments, false, get_properties_by_path_method_handler, gate_);
+
+                auto status = nmos::fields::nc::status(result);
+                auto code = (nc_method_status::ok == status || nc_method_status::property_deprecated == status) ? status_codes::OK : status_codes::InternalError;
+                set_reply(res, code, result);
+            }
+            else
+            {
+                // resource not found for the role path
+                set_error_reply(res, status_codes::NotFound, U("Not Found; ") + role_path);
+            }
+
+            return pplx::task_from_result(true);
+        });
+
+        configuration_api.support(U("/rolePaths/") + nmos::patterns::rolePath.pattern + U("/bulkProperties/?"), methods::OPTIONS, [&model, validate_set_properties_by_path_method_handler, &gate_](http_request req, http_response res, const string_t&, const route_parameters& parameters)
+        {
+            const auto role_path = parameters.at(nmos::patterns::rolePath.name);
+
+            auto lock = model.read_lock();
+            auto& resources = model.control_protocol_resources;
+
+            const auto& resource = details::find_resource(resources, role_path);
+            if (resources.end() != resource)
+            {
+                return details::extract_json(req, gate_).then([res, resources, resource, validate_set_properties_by_path_method_handler, &gate_](value body) mutable
+                {
```

Comment:
Do JSON validation on the extracted body before use

Reply:
OK - will add validation of body before use.